src/model.py

```
@@ -349,6 +348,87 @@ def backward(ctx, grad_output):
+    """
+    VRAM-optimized Fused Head operator.
+    
+    Key Features:
+    1. Forward: Calculates CrossEntropy in chunks to avoid storing full Logits [B*T, V].
+    2. Backward: Recomputes Logits in chunks, decoupling VRAM usage from sequence length.
+    3. Integration: Built-in support for RWKV-specific L2Warp regularization.
+    """
+    @staticmethod
+    def forward(ctx, hidden_states, weight, target, l2warp_factor=1e-4):
+        # hidden_states: [B_T, H], weight: [V, H], target: [B_T]
+        B_T, H = hidden_states.shape
```

Ruff (0.15.6)

[warning] Unpacked variable `H` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)